UnitTests/HttpInterface/api-document-create-spec.rb

```diff
@@ -845,8 +845,204 @@
+      before do
+        @cn = "UnitTestsCollectionUnsynced"
+        @cid = ArangoDB.create_collection(@cn, false)
+      end
+
+      after do
+        ArangoDB.drop_collection(@cn)
+      end
+
+      it "replace a document by _key" do
+        cmd = "/_api/document?collection=#{@cn}"
+        body = "{ \"Hallo\" : \"World\" }"
+        doc = ArangoDB.log_post("#{prefix}-accept", cmd, :body => body, :headers => {})
+
+        doc.code.should eq(202)
+        doc.headers['content-type'].should eq("application/json; charset=utf-8")
+
+        etag = doc.headers['etag']
+        etag.should be_kind_of(String)
+
+        location = doc.headers['location']
+        location.should be_kind_of(String)
+
+        rev = doc.parsed_response['_rev']
+        rev.should be_kind_of(String)
+
+        did = doc.parsed_response['_id']
+        did.should be_kind_of(String)
+
+        key = doc.parsed_response['_key']
+        key.should be_kind_of(String)
+
+        match = didRegex.match(did)
+
+        match[1].should eq("#{@cn}")
+
+        etag.should eq("\"#{rev}\"")
+        location.should eq("/_db/_system/_api/document/#{did}")
+
+        cmd = "/_api/document?collection=#{@cn}&overwrite=true&waitForSync=false"
+        body = "{ \"_key\" : \"#{key}\",  \"Hallo\" : \"ULF\" }"
+        newdoc = ArangoDB.log_post("#{prefix}-accept", cmd, :body => body, :headers => {})
+
+        newrev = newdoc.parsed_response['_rev']
+        newrev.should be_kind_of(String)
+        newrev.should !eq(rev)
+
+        newoldrev = newdoc.parsed_response['_oldRev']
+        newoldrev.should be_kind_of(String)
+        newoldrev.should eq(rev)
+
+        newkey = newdoc.parsed_response['_key']
+        newkey.should be_kind_of(String)
+        newkey.should eq(key)
+        newdoc.code.should eq(202)
+
+        ArangoDB.delete(location)
+        ArangoDB.size_collection(@cn).should eq(0)
+      end
+
+      it "replace a document by _key, return new / old" do
+        cmd = "/_api/document?collection=#{@cn}"
+        body = "{ \"Hallo\" : \"World\" }"
+        doc = ArangoDB.log_post("#{prefix}-accept", cmd, :body => body, :headers => {})
+
+        doc.code.should eq(202)
+        doc.headers['content-type'].should eq("application/json; charset=utf-8")
+
+        etag = doc.headers['etag']
+        etag.should be_kind_of(String)
+
+        location = doc.headers['location']
+        location.should be_kind_of(String)
+
+        rev = doc.parsed_response['_rev']
+        rev.should be_kind_of(String)
+
+        did = doc.parsed_response['_id']
+        did.should be_kind_of(String)
+
+        key = doc.parsed_response['_key']
+        key.should be_kind_of(String)
+
+        match = didRegex.match(did)
+
+        match[1].should eq("#{@cn}")
+
+        etag.should eq("\"#{rev}\"")
+        location.should eq("/_db/_system/_api/document/#{did}")
+
+        cmd = "/_api/document?collection=#{@cn}&overwrite=true&returnNew=true&returnOld=true&waitForSync=true"
```

Comment:
"returnOld" is set here, but the test does not check whether the old document is part of the result.

Reply:
in 967 the old document is inspected